### ISSUE TITLE
Feature Request: YAML Front Matter

### ISSUE THREAD

**Issue by bobrocke** [bobthecow/Markdown.mode#5]
Would you consider adding a state for YAML front matter than pulls in the Coda native YAML mode?

**Comment by bobthecow** [comment]
Yeah, that'd make a lot of sense. I don't have a ton of time to work on it right now, but I'd be happy to accept a pull request if you can get to it before I do :)

**Comment by bobrocke** [comment]
@bobthecow, I downloaded your mode file and made some changes that make it much better. But I'm a GitHub noob. Can you point me to something that can teach me how to turn that into a pull request for you?

**Comment by bobthecow** [comment]
Awesome!

GitHub has a guide here:  https://guides.github.com/activities/contributing-to-open-source/

… but it's probably more complex than you need in this case. The easiest way is simply to click the edit button on the top right corner of the file on GitHub, paste your changes, and fill out the "suggest an edit" form at the bottom. 

This might be out of date now, but here is the blog post where they announce this feature:

https://github.com/blog/844-forking-with-the-edit-button

**Comment by bobrocke** [comment]
Dang! Looking deeper into this, I ended up make a lot of changes, mostly for my own tastes, in addition to adding support for YAML front matter. I'm afraid it's going to me trouble for you to pick out what I've done or for me to roll it back to simply the YAML changes.

I'll push the whole thing and make a pull request, but will fully understand if you ignore it for the mess I've made to it.
